packages/@react-spectrum/numberfield/src/NumberField.tsx

```diff
@@ -146,9 +150,31 @@ const NumberFieldInput = React.forwardRef(function NumberFieldInput(props: Numbe
         style={style}
         className={className}>
         <TextFieldBase
-          UNSAFE_className={classNames(stepperStyle, 'spectrum-Stepper-field')}
+          UNSAFE_className={
+            classNames(
+              stepperStyle,
+              'spectrum-Stepper-field',
+              classNames(inputgroupStyles,
+                'spectrum-InputGroup-field'
+              )
+            )
+          }
           isQuiet={isQuiet}
-          inputClassName={classNames(stepperStyle, 'spectrum-Stepper-input')}
+          inputClassName={
+            classNames(
+              stepperStyle,
+              'spectrum-Stepper-input',
+              classNames(inputgroupStyles,
+                'spectrum-InputGroup-input'
```

Comment:
do we want the right border of the input to be 0 when readonly (this comes from `spectrum-InputGroup-input`)? Perhaps the button's border color should be still be visible?

![image](https://user-images.githubusercontent.com/9661127/107590205-6d7f1200-6bbc-11eb-9f72-5843a0f055e5.png)

Reply:
I opted to follow combobox' lead on this one, might be worth talking to them about